### PR TITLE
add this fix for newer version of pytables. 

### PR DESCRIPTION
**Pull request by skoudoro**
This request resolve and refer to the issue #1142

### CODE REVIEW

**Comment by Garyfallidis** [comment]
@skoudoro  please check the errors reported by Travis https://travis-ci.org/nipy/dipy/jobs/173770112

**Comment by Garyfallidis** [comment]
Actually, I believe you don't even have to have two variables. One would be enough. Just check if tables is greater than 3 (using LooseVersion). One definition TABLES3 should be enough. No need for PY in front. Let me know if this suggestion is clear.  Thanks in advance @skoudoro.

**Comment by skoudoro** [comment]
Thanks for the suggestion @Garyfallidis . Everything is clear, but i still think that my last commit will have a problem on travis test with `tables.__version__` when tables package is not available. what is the good way for optional package ?

**Comment by Garyfallidis** [comment]
Yes, correct. You need to skip the relevant test when `has_tables` is False.

**Comment by Garyfallidis** [comment]
The test is still trying to import `have_tables` but now you call it `has_tables`
```
from dipy.io.dpy import Dpy, have_tables
ImportError: cannot import name have_tables
```

**Comment by skoudoro** [comment]
ok, sorry... I will configure my local environment correctly and run test locally instead of committing stupid thing...

File: dipy/io/dpy.py [hunk 1]
```diff
@@ -39,50 +48,60 @@ def __init__(self, fname, mode='r', compression=0):
...
-        >>> dpw.write_track(B)
-        >>> dpw.write_track(C)
-        >>> dpw.close()
-        >>> dpr = Dpy(fname,'r')
-        >>> A=dpr.read_track()
-        >>> B=dpr.read_track()
-        >>> T=dpr.read_tracksi([0,1,2,0,0,2])
-        >>> dpr.close()
-        >>> os.remove(fname) #delete file from disk
+        >>> def dpy_example():
```
Comment (Garyfallidis) [inline dipy/io/dpy.py]: The indentation inside the function here seems incorrect. It should be 4 spaces or go back to previous version.

**Comment by Garyfallidis** [comment]
Hi @skoudoro Travis looks happy. Check my last comment above on indentation and then I can merge this.

File: dipy/io/dpy.py (hunk 1 above)
Comment (skoudoro) [inline dipy/io/dpy.py]: ok. previous one was wrong (5 spaces cause an indentationError) so a new commit below.

**Comment by arokem** [comment]
OK. I just tested this (including doctest), both with pytables > 3 and pytables < 3. Seems to work fine. Merging.

**Comment by skoudoro** [comment]
Thank you @arokem !  So I close issue #1142

**Comment by arokem** [comment]
Thank you @skoudoro for your work on this!

**Pull request merged**